fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -3336,33 +3345,33 @@ public CompletableFuture<Void> updateRecordCountStateAsync(@Nonnull RecordMetaDa
     }
 
     /**
-     * Set a special store state. A logStamp is required, note that a timestamp will be added automatically.
-     * @param state the new special state
+     * Set a store lock state. A logStamp is required, note that a timestamp will be added automatically.
+     * @param state the new lock state
      * @param logStamp a free text message that can hint future observers about the reasons for setting this state.
```

Comment:
```suggestion
     * Set a store lock state. A reason is required, note that a timestamp will be added automatically.
     * @param state the new lock state
     * @param reason a free text message that can hint future observers about the reasons for setting this state.
```

Also, in the javadoc it notes that the reason is required, but it's just required to be Nonnull, an empty string would be allowed. I'm not sure we want to enforce that, but saying that it's required might be redundant.

Reply:
✅